pkg/apis/istio/v1alpha2/istiocontrolplane_types.proto

```diff
@@ -166,6 +166,11 @@ package v1alpha2;
 
 // IstioControlPlane is a CustomResourceDefinition (CRD) describing an Istio control plane.
 message IstioControlPlane {
+    // ObjectMeta is v11.ObjectMeta.
+    TypeObjectMeta object_meta = 1;
```

Comment:
Do we need to set `json_name` option for `ObjectMeta`? because its json key is `metadata` instead of `objectMeta`.

Reply:
yes, i think that also explains the problem you're seeing with unmarshaling the finalizer field.